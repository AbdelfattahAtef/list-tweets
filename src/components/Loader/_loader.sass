@import "../../styles/_variables.sass"
@import url('https://fonts.googleapis.com/css?family=ZCOOL+QingKe+HuangYou')

.loader
  width: 100%
  height: 100vh
  position: fixed
  background-color: rgba(255,255,255, 0.9)
  top: 0
  left: 0
  z-index: 999999
  display: flex
  justify-content: center
  align-items: center
  flex-direction: column
  &__message
    font-size: $text--large-size
    font-weight: $text--weight-bold
    color: $color--light-blue
    font-family: 'ZCOOL QingKe HuangYou', cursive
  .spinner
    -webkit-animation: spin 1s linear infinite
      animation: spin 1s linear infinite
    border: 3px solid $color--light-blue
    border-top: 3px solid $color--yellow
    border-radius: 50%
    height: 75px
    width: 75px
    margin-bottom: 20px
  span
    font-size: 25px
    display: block
    margin-top: 20px


@-webkit-keyframes spin
  to
    border-top-color: $color--yellow
    -webkit-transform: rotate(360deg)
    transform: rotate(360deg)



@keyframes spin
  to
    border-top-color: $color--yellow
    -webkit-transform: rotate(360deg)
    transform: rotate(360deg)



