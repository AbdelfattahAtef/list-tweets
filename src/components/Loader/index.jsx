import React from "react";
import "./_loader.sass";

function Loader() {
  return (
    <div className="loader">
      <div className="spinner" />
      <div className="loader__message">Loading...</div>
    </div>
  );
}

export default Loader;
