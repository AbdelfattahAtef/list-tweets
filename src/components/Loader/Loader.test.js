import React from 'react';
import Loader from '../Loader';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<Loader />', () => {
    it('should render Loader component', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper.find('div').length).toBe(3);
        expect(wrapper.find('.loader').length).toBe(1);
        expect(wrapper.find('.loader__message').text()).toBe('Loading...');
    });
    it('matches the snapshot', () => {
        const tree = shallow(<Loader />);
        expect(toJson(tree)).toMatchSnapshot();
    })
});