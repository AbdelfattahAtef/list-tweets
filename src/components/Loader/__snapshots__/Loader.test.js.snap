// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> matches the snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="spinner"
  />
  <div
    className="loader__message"
  >
    Loading...
  </div>
</div>
`;
